### DISCUSSION TITLE
Can't get Windows bat to retrieve variable option:COLOR

### DISCUSSION THREAD

**Discussion by Txori** [ImageMagick/ImageMagick#3981] [Help] [answered]
Hi, I discovered ImageMagick today, and I've been already stuck several hours with that problem:
I can't retrieve my variable option:COLOR. Funny thing is that filename:FILE is working perfectly...

Here's my script:
```
magick convert ^
 *.jpeg ^
 -set filename:FILE "%%[t]" ^
 ( +clone ^
   -resize 1x1 ^
   -set option:COLOR "%%[pixel:p{0,0}]" ^
   +delete ^
 ) ^
 -bordercolor "%%[COLOR]" ^
 -border x500 ^
 -resize 1024 ^
 -gravity center ^
 -crop 1024x1024+0+0 ^
 "%%[filename:FILE].png"
```

Unfortunately, -bordercolor "%%[COLOR]" isn't working. I tried a lot of things. All I get is
`convert: unrecognized color '%[COLOR]' @ warning/color.c/GetColorCompliance/1054.`

So right now, I'm using that instead:
` -bordercolor "#34373C"`
It works, but it's not very interactive.

Thanks for your help!

**Comment by fmw42** [comment]
What platform and what version of ImageMagick. If not on Window, do not double %. Also in ImageMagick 7, use magick not magick convert.

**Comment by snibgo** [comment]
@Txori: Your script is bad. A magick command should start with the word "magick", or "convert" if it is an old script.

I suspect you haven't used "magick", but either "convert" or "magick convert".

**Comment by Txori** [comment]
It's Windows 10 and IM7.
I used "magick convert", yes... Don't know why it wasn't showing up in my post. (I edited it)
After suppressing "convert", my COLOR variable is now working perfectly. Thank you very much !

**Comment by Txori** [comment]
After more tests, I simplified the script to this:
```
magick ^
 ./todo/*.* ^
 -set filename:FILE "%%[t]" ^
 -format png ^
 -size 1024x1024 ^
 -background "%%[pixel:p{0,0}]" ^
 -gravity center ^
 -resize 1024 ^
 -extent 1024x1024 ^
 "./done/%%[filename:FILE].png"
```

My new problem is that COLOR is always black instead of the color at 0,0

I also tried hex, but the result is always black...
` -background "#%%[hex:p{0,0}]" ^`

**Comment by snibgo** [comment, marked as answer]
You are setting the background colour to the pixel at (0,0) of the _last_ image. You don't get a different background colour for each image.

I suggest creating a command that processes one file at a time, not all at once. Then put that inside a "for" loop.

**Reply by Txori** [reply]
Oh thanks, I didn't know that the last color picker was in fact the only one taken in account...

Here's my new WORKING batch script :
```
for /R %%f in (todo\*.*) do (

echo process "%%f"

magick ^
 "./todo/%%f" ^
 -set filename:FILE "%%[t]" ^
 -background "#%%[hex:p{2,2}]" ^
 -format PNG ^
 -gravity center ^
 -size 1024x1024 ^
 -resize 1024 ^
 -extent 1024x1024 ^
 "./done/%%[filename:FILE].png"

)
```

**Comment by snibgo** [comment]
Good stuff. Note that your "-size 1024x1024" does nothing. It has no effect, and can be removed.

**Reply by Txori** [reply]
Ok, thanks :)
